ts/components/wallet/OutcomeCodeMessageComponent.tsx

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import { ImageSourcePropType, SafeAreaView } from "react-native";
+import I18n from "../../i18n";
+import View from "../ui/TextWithIcon";
+import { emptyContextualHelp } from "../../utils/emptyContextualHelp";
+import { IOStyles } from "../core/variables/IOStyles";
+import BaseScreenComponent from "../screens/BaseScreenComponent";
+import { InfoScreenComponent } from "../infoScreen/InfoScreenComponent";
+import { renderInfoRasterImage } from "../infoScreen/imageRendering";
+import FooterWithButtons from "../ui/FooterWithButtons";
+import { cancelButtonProps } from "../../features/bonus/bonusVacanze/components/buttons/ButtonConfigurations";
+import { OutcomeCode } from "../../types/outcomeCode";
+
+type OwnProp = {
+  outcomeCode: OutcomeCode;
+  successComponent: React.FC;
+  successFooter: React.FC;
+  onClose: () => void;
+};
+type Props = OwnProp;
+
+const blockingFooterWithButton = (onClose: () => void) => (
+  <FooterWithButtons
+    type={"SingleButton"}
+    leftButton={cancelButtonProps(
+      onClose,
+      I18n.t("wallet.outcomeMessage.cta.close")
+    )}
+  />
+);
+
+/**
+ * This component renders the message associated to the outcome code of a payment.
+ * This component renders an image, a title and a description or a success component.
+ *
+ * Since there are 3 possible case for the outcomeCode.status and the footer buttons are
+ * related to every case the choice of the footer is based on the outcomeCode.status props.
+ *
+ * @param props
+ */
+const OutcomeCodeMessageComponent: React.FC<Props> = (props: Props) => {
+  const locale = I18n.currentLocale() === "en" ? "en-EN" : "it-IT";
+  const title = props.outcomeCode.title ? props.outcomeCode.title[locale] : "";
```

Comment:
Could it be title and description `undefined`?

`const title = props.outcomeCode.title ? props.outcomeCode.title[locale] : undefined;`

So in this way we can use them to have a conditional rendering

```typescript
{title && description && (
              <InfoScreenComponent
                image={renderInfoRasterImage(
                  props.outcomeCode.icon as ImageSourcePropType
                )}
                title={title}
                body={description}
              />
            )}
```

Reply:
Yes, they could.
I choose to conditional render the component checking only the **title** since the description is an optional field.
Modified in [c3bb625](https://github.com/pagopa/io-app/pull/2842/commits/c3bb625596ddddd2a2e60c5da7e26a86c840a503)